Form/DataTransformer/ProviderDataTransformer.php

```diff
@@ -32,9 +38,10 @@ class ProviderDataTransformer implements DataTransformerInterface
      * @param string $class
      * @param array  $options
      */
-    public function __construct(Pool $pool, $class, array $options = array())
+    public function __construct(Pool $pool, $class, LoggerInterface $logger, array $options = array())
```

Comment:
Please put that argument last for BC's sake, and make it optional. When it is null, detect it and assign a `NullLogger` instance to `$logger`